funnel/views/account.py

```diff
@@ -91,6 +93,46 @@ def account(self):
     def saved(self):
         return {'saved_projects': current_auth.user.saved_projects}
 
+    @route('siteadmin/comments', endpoint='siteadmin_comments', methods=['GET', 'POST'])
+    @requires_login
+    @render_with('siteadmin_comments.html.jinja2')
+    def siteadmin_comments(self):
+        if not (
+            current_auth.user.is_comment_moderator
+            or current_auth.user.is_user_moderator
+        ):
+            return redirect('/')
```

Comment:
abort(403)

Reply:
There's a hack for this, actually. Use a ModelView with `model = User` and set `loader()` to simply return `current_auth.user`. That way you can use `@requires_roles`

Reply:
Bad hack as we're not operating on the User (unlike in /account views). Better to create a non-persistent `Site` object that implements `RoleMixin`.